File: C#/100_Same_Tree.cs
using System;

namespace LeetCode
{
    // Definition for a binary tree node.
    public class TreeNode {
        public int val;
        public TreeNode left;
        public TreeNode right;
        public TreeNode(int x) { val = x; }
      }
    public class Solution
    {
        public bool IsSameTree(TreeNode p, TreeNode q)
        {
            if (p == null && q == null)
                return true;
            else if (p == null || q == null)
                return false;
            else
            {
                if (p.val == q.val)
                {
                    if (IsSameTree(p.left, q.left))
                        return IsSameTree(p.right, q.right);
                }
                return false;
            }
        }
        static void Main(string[] args)
        {
            Solution s = new Solution();
            Console.Write(s.IsSameTree(new TreeNode(1), new TreeNode(2)));
        }
    }
}

File: Python/100_Same_Tree.py
# Definition for a binary tree node.
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution(object):
    def isSameTree(self, p, q):
        """
        :type p: TreeNode
        :type q: TreeNode
        :rtype: bool
        """
        if p is None and q is None:
            return True
        elif p is None or q is None:
            return False
        else:
            if p.val == q.val:
                if self.isSameTree(p.left, q.left):
                    return self.isSameTree(p.right, q.right)
            return False


def test():
    s = Solution();
    print s.isSameTree(TreeNode(1), TreeNode(2))


if __name__ == '__main__':
    test()
